src/device_trezor/device_trezor_base.cpp

```diff
@@ -382,8 +385,6 @@ namespace trezor {
       } else {
         m_device_session_id.clear();
       }
-
-      initMsg->release_session_id();
```

Comment:
This causes memory corruption. If exception is thrown, this release was not called. Destructor of the initMsg then called `delete` on locally defined `tmp_session_id`.